app/models/external_activity.rb

```diff
@@ -3,6 +3,7 @@ class ExternalActivity < ActiveRecord::Base
 
   # see https://github.com/sunspot/sunspot/blob/master/README.md
   searchable do
+    integer :id
```

Comment:
Sunspot already adds the model id to the search result. Perhaps it isn't easy to access?  It is stored in an id field of Solr with a form like `ExternalActivity 123` (ditto for Interactive, and Investigation)